stdlib/LinearAlgebra/docs/src/index.md

```diff
@@ -306,6 +308,52 @@ of the Linear Algebra documentation.
 | `Schur`            | [Schur decomposition](https://en.wikipedia.org/wiki/Schur_decomposition)                                       |
 | `GeneralizedSchur` | [Generalized Schur decomposition](https://en.wikipedia.org/wiki/Schur_decomposition#Generalized_Schur_decomposition) |
 
+## [Orthogonal matrices (`AbstractQ`)](@id man-linalg-abstractq)
+
+Some matrix factorizations generate orthogonal/unitary "matrix" factors. These
+factorizations include QR-related factorizations obtained from calls to [`qr`](@ref), i.e.,
+`QR`, `QRCompactWY` and `QRPivoted`, the Hessenberg factorization obtained from calls to
+[`hessenberg`](@ref), and the LQ factorization obtained from [`lq`](@ref). While these
+orthogonal/unitary factors admit a matrix representation, their internal representation
+is, for performance and memory reasons, different. Hence, they should be rather viewed as
+matrix-backed, function-based linear operators. In particular, reading, for instance, a
+column of its matrix representation requires running "matrix"-vector multiplication code,
+rather than simply reading out data from memory (possibly filling parts of the vector with
+structural zeros). Another clear distinction from other, non-triangular matrix types is
+that the underlying multiplication code allows for in-place modification during multiplication.
```

Comment:
I think it should be mentioned that `AbstractQ` can behave like a square or a rectangular matrix depending on context:
```julia
julia> Q = qr(randn(5,3)).Q
5×5 LinearAlgebra.QRCompactWYQ{Float64, Matrix{Float64}, Matrix{Float64}}:
 -0.138692  -0.143851  -0.462938     0.851611   0.143241
  0.249145  -0.663384   0.642309     0.253204   0.145518
  0.877971   0.426824   0.00364718   0.20546    0.0689985
 -0.355541   0.564291   0.604996     0.396391  -0.178943
 -0.146453   0.196534   0.0842246   -0.106333   0.959961

julia> Matrix(Q)
5×3 Matrix{Float64}:
 -0.138692  -0.143851  -0.462938
  0.249145  -0.663384   0.642309
  0.877971   0.426824   0.00364718
 -0.355541   0.564291   0.604996
 -0.146453   0.196534   0.0842246

julia> Q * randn(5)
5-element Vector{Float64}:
 -0.27087037277913545
  0.8289729117870301
  0.9710403448488311
  1.07265126537851
  1.5917990225008223

julia> Q * randn(3)
5-element Vector{Float64}:
 -0.005839962023632572
 -0.09405929877277669
  0.2543702202805215
 -0.06322507238957749
 -0.014939190731340635
```

Reply:
Thanks for the review! I thought about mentioning this, but I'm not exactly sure whether this should be related to `AbstractQ` in general, or to the specific types coming from LAPACK's `qr`. For instance, I realized that the `Q` from `lq` (I think it's called `LQPackedQ`) did not subtype `AbstractQ` because it has this flexible size only when it's multiplied from the right (or its adjoint from the left), in contrast to the Hessenberg and QR Q's. Perhaps I could mention this behaviour, mention that this is a special feature of some Q's and link to the `qr` docstring?